tartufo/cli.py

```diff
@@ -134,56 +134,29 @@ def main(ctx: click.Context, **kwargs: config.OptionTypes) -> None:
     pre-commit hook.
     """
     options = types.GitOptions(**kwargs)  # type: ignore
-    if not any((options.entropy, options.regex)):
-        util.fail("No analysis requested.", ctx)
+
     if not any((options.pre_commit, options.repo_path, options.git_url)):
         util.fail("You must specify one of --pre-commit, --repo-path, or git_url.", ctx)
 
-    rules_regexes: Dict[str, Pattern] = {}
-    if options.regex:
-        try:
-            rules_regexes = config.configure_regexes(
-                options.default_regexes,
-                options.rules,
-                options.git_rules_repo,
-                options.git_rules_files,
-            )
-        except ValueError as exc:
-            util.fail(str(exc), ctx)
-        if not rules_regexes:
-            util.fail("Regex checks requested, but no regexes found.", ctx)
-
-    # read & compile path inclusion/exclusion patterns
-    path_inclusions: List[Pattern] = []
-    path_exclusions: List[Pattern] = []
-    if options.include_paths:
-        path_inclusions = config.compile_path_rules(options.include_paths.readlines())
-    if options.exclude_paths:
-        path_exclusions = config.compile_path_rules(options.exclude_paths.readlines())
-
     found_issues: List[scanner.Issue] = []
     remove_repo = False
     repo_path = str(options.repo_path)
-    if options.pre_commit:
-        found_issues = scanner.find_staged(
-            project_path=str(options.repo_path),
-            options=options,
-            custom_regexes=rules_regexes,
-            path_inclusions=path_inclusions,
-            path_exclusions=path_exclusions,
-        )
-    else:
-        if options.git_url:
-            repo_path = util.clone_git_repo(options.git_url)
-            remove_repo = True
+    repo_scanner: scanner.ScannerBase
```

Comment:
Good callout, but in this case no. Rather than a variable assignment, this is a [PEP 526](https://www.python.org/dev/peps/pep-0526/) variable annotation. This is just a way of telling all of the tooling what type this variable is. A lot of times this can be inferred, but in this case `mypy` will actually throw a typing conflict if this isn't declared here, because below we assign both `repo_scanner = GitPreCommitScanner(...)` and `repo_scanner = GitRepoScanner(...)`, and it can't figure out what type the variable is actually supposed to be.